docs/side_quests/splitting_and_grouping.md

```diff
@@ -600,11 +639,14 @@ To do so, first we define the closure as a new variable:
 
 === "After"
 
-    ```groovy title="main.nf" linenums="2" hl_lines="4"
+    ```groovy title="main.nf" linenums="2" hl_lines="7"
     ch_samplesheet = Channel.fromPath("./data/samplesheet.csv")
-                        .splitCsv(header: true)
+                    .splitCsv(header: true)
+                    .map{ row ->
+                        [[id:row.id, repeat:row.repeat, type:row.type], row.bam]
+                    }
 
-    getSampleIdAndReplicate = { sample -> [ sample.subMap(['id', 'repeat']), sample ] }
+    getSampleIdAndReplicate = { meta, file -> [ meta.subMap(['id', 'repeat']), meta, file ] }
```

Comment:
```suggestion
    getSampleIdAndReplicate = { meta, file -> [ meta.subMap(['id', 'repeat']), meta, file(file) ] }
```

Maybe we should actually make this a path object at this point, and point to the files side-quest?